keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -0,0 +1,428 @@
+
+## Motivation
+
+File and block are first class citizens within the Kubernetes ecosystem.  Object, though very different under the hood, is a popular means of storing data, especially against very large data sources.   As such, we feel it is in the interest of the community to integrate object storage into Kubernetes, supported by the SIG-Storage community.  In doing so, we can provide Kubernetes cluster users and administrators a normalized and familiar means of managing object storage. 
+
+While absolute portability cannot be guaranteed because of incompatibilities between providers, workloads reliant on a given protocol (e.g. one of S3, GCS, Azure Blob) may be defined in a single manifest and deployed wherever that protocol is supported.
+
+This proposal does _not_ include a standardized *protocol* or abstraction of storage vendor APIs
+
+## Goals
+
++ Specify object storage Kubernetes APIs for the purpose of orchestrating object store operations
++ Implement a Kubernetes controller architecture with support for pluggable provisioners
++ As MVP, be accessible to the largest groups of consumers by supporting the major object storage protocols (S3, Google Cloud Storage, Azure Blob) while being extensible for future protocol additions.
++ Present similar workflows for both greenfield and brownfield bucket operations.
+
+## Non-Goals
+
++ Define the _data-plane_ object store interface to replace or supplement existing vendor interfaces (i.e. replace GCS, S3, or Azure Blob)
+
+##  Vocabulary
+
++  _Brownfield Bucket_ - a storage instance created out of band.
++ _BucketRequest_ - A user-namespaced custom resource representing a request for a storage instance endpoint.
++  _BucketClass_ - A cluster-scoped custom resource containing fields defining the provisioner and an immutable parameter set for creating new buckets
++ _Bucket_ - A cluster-scoped custom resource referenced by a Bucket and containing connection information and metadata for a storage instance.
++ _Greenfield Bucket_ - a bucket created by automation.
++  _Object_ - An atomic, immutable unit of data stored in buckets.
++  *Storage Instance* - Refers to the back object storage endpoint being abstracted by the Bucket API (a.k.a “bucket” or “container”).
++ _Driverless_ - A system where not driver is deployed to automate object store operations.  COSI automation may still be deployed to managed COSI APIs.  See [Architecture](#architecture).
```

Comment:
"A system where no driver is ..."

Reply:
Driverless was used as a synonym for "static brownfield" and there is a reference to static provisioning in the  BucketClass.provisioner description. But this concept is not mentioned anywhere else in the KEP. Maybe we are not longer supporting static/driverless for MVP??  @wlan0